pkg/trace/sampler/probabilistic.go

```diff
@@ -0,0 +1,148 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022-present Datadog, Inc.
+
+package sampler
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"hash/fnv"
+	"strconv"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/proto/pbgo/trace"
+	"github.com/DataDog/datadog-agent/pkg/trace/config"
+	"github.com/DataDog/datadog-agent/pkg/trace/log"
+	"github.com/DataDog/datadog-agent/pkg/trace/watchdog"
+	"github.com/DataDog/datadog-go/v5/statsd"
+)
+
+const (
+	// These constants exist to match the behavior of the OTEL probabilistic sampler.
+	// See: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/6229c6ad1c49e9cc4b41a8aab8cb5a94a7b82ea5/processor/probabilisticsamplerprocessor/tracesprocessor.go#L38-L42
```

Comment:
Unfortunately we're taking a risk here since that spec is still a v0 and might change in the future.

Reply:
Yes, indeed. Unfortunately much of otel is still v0, and we regularly have issues with changing APIs.